GherkinSpec.TestAdapter/TestDiscoverer.cs

```diff
@@ -28,12 +32,13 @@ public class TestDiscoverer : ITestDiscoverer
             }
         }
 
-        public static IEnumerable<TestCase> DiscoverTests(IEnumerable<string> sources, IMessageLogger logger)
+        public IEnumerable<TestCase> DiscoverTests(IEnumerable<string> sources, IMessageLogger logger)
         {
             foreach (var source in sources)
             {
                 foreach (var testCase in DiscoverTests(source, logger))
                 {
+                    EnsureTestCaseNameIsUnique(testCase);
```

Comment:
I think we may need to only ensure names are unique _within the same source_.  It adds weight to your question about whether we should move the two private fields and name-check logic into a separate class.  We could then in this method use a fresh instance for each test source?

Reply:
Thanks, extracted this method & fields to `UniqueTestCaseNameResolver` & instantiated in the scope of each source.